src/highdicom/map/sop.py

```diff
@@ -0,0 +1,732 @@
+              for multiple 2D frames at different spatial image positions and
+              the array shall have shape ``(n, r, c)``, where ``n`` is the
+              number of frames, ``r`` is the number of rows per frame, and
+              ``c`` is the number of columns per frame.
+
+            * In case of a 4D array, the values represent multiple channels
+              for multiple 2D frames at different spatial image positions and
+              the array shall have shape ``(n, r, c, m)``, where ``n`` is the
+              number of frames, ``r`` is the number of rows per frame, ``c`` is
+              the number of columns per frame, and ``m`` is the number of
+              channels.
+
+        series_instance_uid: str
+            UID of the series
+        series_number: Union[int, None]
+            Number of the series within the study
+        sop_instance_uid: str
+            UID that should be assigned to the instance
+        instance_number: int
+            Number that should be assigned to the instance
+        manufacturer: str
+            Name of the manufacturer (developer) of the device (software)
+            that creates the instance
+        manufacturer_model_name: str,
+            Name of the model of the device (software)
+            that creates the instance
+        software_versions: Union[str, Tuple[str]]
+            Versions of relevant software used to create the data
+        device_serial_number: str
+            Serial number (or other identifier) of the device (software)
+            that creates the instance
+        contains_recognizable_visual_features: bool
+            Whether the image contains recognizable visible features of the
+            patient
+        real_world_value_mappings: Union[Sequence[highdicom.map.RealWorldValueMapping], Sequence[Sequence[highdicom.map.RealWorldValueMapping]]
+            Descriptions of how stored values map to real-world values.
+            The concept of real-world values is a bit fuzzy or the mapping may
+            be difficult to describe (e.g., in case of a transformation
+            performed by a deep convolutional neural network).
+            The real-world value mapping may simply describe an identity
+            function that maps stored values to unit-less real-world values.
+            Each channel encoded in `pixel_array` shall be described with one
+            or more real-world value mappings.
+            If `pixel_array` is a 2D or 3D array and only one channel exists
+            at each spatial image position), then one or more real-world value
+            mappings shall be provided in a flat sequence.
+            If `pixel_array` is a 4D array and multiple channels exist at each
+            spatial image position, then one or more mappings shall be provided
+            for each channel in a nested sequence of length ``m``, where ``m``
+            shall match the channel dimension of the `pixel_array``.
```

Comment:
```suggestion
            Descriptions of how stored values map to real-world values.
            Each channel encoded in `pixel_array` shall be described with one
            or more real-world value mappings. Multiple mappings might be
            used for different representations such as log versus linear scales
            or for different representations in different units.
            If `pixel_array` is a 2D or 3D array and only one channel exists
            at each spatial image position), then one or more real-world value
            mappings shall be provided in a flat sequence.
            If `pixel_array` is a 4D array and multiple channels exist at each
            spatial image position, then one or more mappings shall be provided
            for each channel in a nested sequence of length ``m``, where ``m``
            shall match the channel dimension of the `pixel_array``.

            In some situations the mapping may be difficult to describe (e.g., in
            case of a transformation performed by a deep convolutional neural
            network). The real-world value mapping may simply describe an
            identity function that maps stored values to unit-less real-world
            values.
```

Swapping around the order here makes this clearer I think. I added a point of clarification about why multiple mappings may be used. I also reworded the last part to have a more formal tone.